au4/src/projectscene/view/timeline/timelinecontext.cpp

```diff
@@ -205,6 +205,18 @@ double TimelineContext::positionToTime(double position, bool withSnap) const
     return result;
 }
 
+double TimelineContext::singleStepToTime(double position, Direction direction) const
+{
+    double result = m_frameStartTime + position / m_zoom;
+    auto viewState = this->viewState();
+
+    if (viewState && viewState->isSnapEnabled().val) {
```

Comment:
I would recommend using isSnapEnabled as an argument and removing the project dependency from TimelineContext